qiskit_experiments/randomized_benchmarking/rb_analysis.py

```diff
@@ -0,0 +1,133 @@
+
+try:
+    from matplotlib import pyplot as plt
+
+    HAS_MATPLOTLIB = True
+except ImportError:
+    HAS_MATPLOTLIB = False
+
+
+class RBAnalysis(CurveFitAnalysis):
+    """RB Analysis class."""
+
+    # pylint: disable = arguments-differ, invalid-name
+    def _run_analysis(
+        self,
+        experiment_data,
+        p0: Optional[List[float]] = None,
+        plot: bool = True,
+        ax: Optional["AxesSubplot"] = None,
+    ):
+        """Run analysis on circuit data.
+        Args:
+            experiment_data (ExperimentData): the experiment data to analyze.
+            p0: Optional, initial parameter values for curve_fit.
+            plot: If True generate a plot of fitted data.
+            ax: Optional, matplotlib axis to add plot to.
+        Returns:
+            tuple: A pair ``(analysis_result, figures)`` where
+                   ``analysis_results`` may be a single or list of
+                   AnalysisResult objects, and ``figures`` may be
+                   None, a single figure, or a list of figures.
+        """
+        # TODO: Get from experiment level metadata
+        num_qubits = len(experiment_data.data[0]["metadata"]["qubits"])
+
+        # Fit function
+        def fit_fun(x, a, alpha, b):
+            return a * alpha ** x + b
+
+        # Initial guess function
+        # NOTE: I don't think this is a good guess function
+        # its just inserted as a place holder for one
+        # pylint: disable = unused-argument
+        def p0_func(xdata, ydata, sigma=None):
+            xmin = np.min(xdata)
+            y_mean_min = np.mean(ydata[xdata == xmin])
+
+            xmax = np.max(xdata)
+            y_mean_max = np.mean(ydata[xdata == xmax])
+
+            b_guess = 1 / (2 ** num_qubits)
+            a_guess = 1 - b_guess
+            alpha_guess = np.exp(
+                np.log((y_mean_min - b_guess) / (y_mean_max - b_guess)) / (xmin - xmax)
+            )
+            # Make sure alpha guess is feasible
+            alpha_guess = max(min(alpha_guess, 1), 0)
+            return [a_guess, alpha_guess, b_guess]
+
+        # Run CurveFitAnalysis
+        analysis_result, figs = super()._run_analysis(
+            experiment_data,
+            fit_fun,
+            p0=p0,
+            p0_func=p0_func,
+            bounds=([0, 0, 0], [1, 1, 1]),
+            fit_mean_data=True,
+            plot=plot,
+            ax=ax,
+        )
+
+        # Add EPC data
+        popt = analysis_result["popt"]
+        popt_err = analysis_result["popt_err"]
+        scale = (2 ** num_qubits - 1) / (2 ** num_qubits)
+        analysis_result["EPC"] = scale * (1 - popt[1])
+        analysis_result["EPC_err"] = scale * popt_err[1] / popt[1]
+        analysis_result["plabels"] = ["A", "alpha", "B"]
+
+        # Format figure
+        if figs is not None:
+            self._format_plot(figs[0], analysis_result)
+            # TODO: figure out what to do with plots
+            plt.show()
```

Comment:
I know it's only a draft and a TODO block, but in any case I think `show()` should be called externally, not as a side effect of `_run_analysis()`

Reply:
This was removed in the analysis overhaul (but figures are still not handled, we'll add this next)